### PR TITLE
api

### PR DESCRIPTION
**Pull request by Friedchicken-42**
## Summary

Added `IngredientApi` and `RecipeApi`
Added dataclasses `Ingredient`, `Recipe` and `RecipeFull`

### CODE REVIEW

File: app/src/main/java/com/example/ecodigify/api/IngredientApi.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+class IngredientApi : Api() {
+    private val url = "https://world.openfoodfacts.net/api/v2/product"
+    private val regex = "\\d+".toRegex()
+
+    suspend fun search(code: String): Ingredient {
+        val out: IngredientApiOutput = client.get("$url/$code?product_type=food").body()
+        println(out)
```
Comment (Friedchicken-42) [inline app/src/main/java/com/example/ecodigify/api/IngredientApi.kt]: Questo volendo potrebbe essere anche loggato

File: app/src/main/java/com/example/ecodigify/dataclass/Ingredient.kt [hunk 2]
```diff
@@ -12,5 +12,6 @@ data class Ingredient(
     val name: String,
     @ColumnInfo(name = "expiration_date")
     val expirationDate: LocalDate,
+    val possible_names: List<String>,
```
Comment (saccarosium) [inline app/src/main/java/com/example/ecodigify/dataclass/Ingredient.kt:15]:
```suggestion
    @ColumnInfo(name = "possible_names")
    val possibleNames: List<String>,
```

File: app/src/main/java/com/example/ecodigify/dataclass/Recipe.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.example.ecodigify.dataclass
+
+data class Recipe(
+    val id: Int,
+    val name: String,
+    val thumb: String,
```
Comment (saccarosium) [inline app/src/main/java/com/example/ecodigify/dataclass/Recipe.kt:6]:
```suggestion
    val thumbnail: Uri,
```

Comment (Friedchicken-42) [reply app/src/main/java/com/example/ecodigify/dataclass/Recipe.kt]: fixed in https://github.com/Friedchicken-42/progettoIngegneriaSoftware/pull/3/commits/64326b6dd375162cdf0190fcbc17c6e6593ff96e

**Pull request merged**